### DISCUSSION TITLE
Raspberry Pi file storage

### DISCUSSION THREAD

**Discussion by anderep** [jasonacox/Powerwall-Dashboard#271] [General]
should i use a usb ssd and mount as Powerwall-Dashboard, or is the microSD sufficient, size and number of writes?
I'm all ready at 44MB in /Powerwall_Dashboard/influxdb/data/powerwall

**Comment by ThePnuts** [comment]
How long have you had it running? I just looked at mine for comparison.

I have 4 strings on a single PW+, not sure of your setup, but this can help give you an idea.

I am netting about 150mb per month total. 
Only looking at only aggregate data, its about 20mb a month.

So if you have the default retention, you would be keeping only about 20mb a month and 130mb would return to free space as the retention clears it each day once at the time limit.

If you get it running, you could make a simple dashboard to see how much its writing, etc. though.

I do not think you would have a problem running completely on an SD card. Just do regular backups if your worried (and test them). I think even the cheapest Class 10 SD would be able to keep up with what I am doing, so just PyPowerwall should be super fine with overhead for something else.


Not directly comparable, but I am logging from 2 servers. their metrics and PyPowerwall data 2x (Once for a more default setup and once for what I did with Influxdb2 and Flux) and Plex metrics, so I am logging a lot. Its logging to a 6 SSD raid 10 array

![image](https://user-images.githubusercontent.com/126296751/236906514-96ae2939-cb15-4891-b36c-4829a76bdc12.png)

**Reply by anderep** [reply]
i got a 8GB RPi4, with 128GB microSD card.  so i should be fine with file system size, i just home influxdb isn't constantly opening files, packing them and then rewriting, hope just adding to file. 
RPi installations put all the scratch files on ram so not to cause the file system to fail. 
I tried installing Powerwall Dashboard on my clearOS server but the install failed setting up the firewall.  it has a raid10 array, 10x 16TB drives.

**Reply by ThePnuts** [reply]
I think you should be more then fine with the Pi. I would keep trying to get it working on your server, though I'm not sure all that that would entail on ClearOS.

**Reply by jasonacox** [reply]
Nice analysis @ThePnuts !  @anderep if you do get it to work with ClearOS, let us know what was required.

**Reply by anderep** [reply]
I didn’t know what ports needed to be opened.   If the install would tell you what needed to be opened and allow choice to continue or not is all that is needed. I don’t think any ports needed to be open to wan, and all ports are open one the lan. I use open vpn to gain access to my network fan wan and don’t open ports.
